### DISCUSSION TITLE
evcc.yamal with 2 x Alfen Eve Pro Single Line, Huawai Sun2000-30KTL-M3, Elgris Smartmeter (SMA Energy Manager), SonnenBatterie 10

### DISCUSSION THREAD

**Discussion by dirk199** [community/community#138037] [Other Feature Feedback, Questions, & Ideas]
I have been trying to set up my 2 Alfen Wallboxes for Solar Charging. Everything i tried faild, so i detected EVCC and tried to set up my RaspberryPi.
After only 2 days i got it connected somehow, but i am not sure if it is correct:

Problem:
1) i can not add my Huawei inverter ? Do I need it to get it running correct? 
2) How can I add the second Wallbox? Both wallboxes have "active load balancing" activated.
3) can i add the sonnen batterie? 
4) I can only choose between "off" and "fast". The button "PV" and "MINPV" is not there? I am not sure if it works correctly with this options. I want to use solar charging 100% ?

My setting:
- Charger: 2 x Alfen Pro Single Line 
- Inverter: Huawei Sun2000-30KTL-M3 Inverter
- Dongle: SDongleA-05
- Batterie: SonnenBatterie 10
- Smartmeter: Elgris SMA1278004612 (Identicl to SMA Engergy Meter ?)

My current evcc.yaml

# open evcc at http://evcc.local:7070
network:
  schema: http
  host: evcc.local # .local suffix announces the hostname on MDNS
  port: 7070

log: debug
levels:
  cache: error

# unique installation id
plant: xyz

meters:
  - name: my_grid
    type: template
    template: cg-em24
    usage: grid
    
# Modbus TCP
    modbus: tcpip
    id: 1
    host: 192.168.178.128 # Hostname
    port: 502 # Port

chargers:
  - name: my_charger
    type: template
    template: alfen
    
    # Modbus TCP
    modbus: tcpip
    id: 1
    host: 192.168.178.122 # Hostname
    port: 502 # Port

vehicles:
  - name: ev1
    title: ID.3
    type: template
    icon: car
    user: xxx.yyy@protonmail.com  
    password: xxxxxxx
    template: vw
    vin: 1234567890  
    capacity: 58  
    phases: 3  
    cache: 15m  
    mode: minpv  
    minCurrent: 6  
    maxCurrent: 16  
    timeout: 10s 

loadpoints:
  - title: Garage # display name for UI
    charger: my_charger # Wallbox Referenz
    mode: pv # charge mode (off, now, minpv, pv)

    # remaining settings are experts-only and best left at default values
    priority: 0 # relative priority for concurrent charging in PV mode with multiple loadpoints (higher values have higher priority)
    soc:
      # polling defines usage of the vehicle APIs
      # Modifying the default settings it NOT recommended. It MAY deplete your vehicle's battery
      # or lead to vehicle manufacturer banning you from API use. USE AT YOUR OWN RISK.
      poll:
        # poll mode defines under which condition the vehicle API is called:
        #   charging: update vehicle ONLY when charging (this is the recommended default)
        #   connected: update vehicle when connected (not only charging), interval defines how often
        #   always: always update vehicle regardless of connection state, interval defines how often (only supported for single vehicle)
        mode: charging
        # poll interval defines how often the vehicle API may be polled if NOT charging
        interval: 60m
      estimate: true # set false to disable interpolating between api updates (not recommended)
    enable: # pv mode enable behavior
      delay: 1m # threshold must be exceeded for this long
      threshold: 0 # grid power threshold (in Watts, negative=export). If zero, export must exceed minimum charge power to enable
    disable: # pv mode disable behavior
      delay: 3m # threshold must be exceeded for this long
      threshold: 0 # maximum import power (W)

interval: 30s # control cycle interval

sponsortoken: xyz

# sponsors can set telemetry: true to enable anonymous data aggregation
# see https://github.com/evcc-io/evcc/discussions/4554
telemetry: false

site:
  title: Pension SiLa
  meters:

On my desktop i see the following:

![Ansicht](https://github.com/user-attachments/assets/f4af6b81-ac81-4846-a8ff-832e90fc5129)

Thank you very much.

Dirk

**Comment by queenofcorgis** [comment]
Hi @dirk199 ,

Thanks for being a part of the GitHub Community, we're glad you're here!

If you're looking for help for this specific topic, you might want to try asking for help somewhere that focuses on these devices. It's possible that another GitHub user might have run into this same issue and can help, but the GitHub Community Discussions focuses primarily on topics related to GitHub itself or collaboration on project development and ideas. We want to make sure you’re getting the best support you can, but this space may not be the right place for this particular topic.

Best of luck!
